### PR TITLE
Docs: Fix custom styling, restore bullets

### LINKED ISSUE #12573: PR #12326 made the bullet lists in the change log invisible
Not only in the change log but globally, too. There are plenty of examples @ https://docs.pytest.org/en/latest/changelog.html#new-features and also in the events widget. What was the intention, with this originally?

@ferdnyc actually already called out this problem in https://github.com/pytest-dev/pytest/pull/12326#issuecomment-2119581848, but it wasn't addressed.

_Originally posted by @webknjaz in https://github.com/pytest-dev/pytest/pull/12326#discussion_r1662635927_

### LINKED ISSUE #12576: Inject CSS via `html_css_files` rather than inline styles in the sidebar
See https://pradyunsg.me/furo/customisation/injecting/#css-or-js for the guidance from the theme for how to inject CSS.

_Originally posted by @pradyunsg in https://github.com/pytest-dev/pytest/pull/12326#discussion_r1666516622_

### PR DESCRIPTION
**Pull request by ferdnyc**
- Move the custom CSS to a dedicated `pytest.css` file in `_static/`, and reference it from the configuraton's `html_css_files` list.
- Update the list-customization styling to only affect `.sidebar-scroll` (the navigation sidebar), so that bullets are visible in the rest of the documentation.
- Remove the list indent from the navigation sidebar, but only at the top level, so that navigation is still nested with visible indents.

Fixes #12573
Fixes #12576 
Closes #12575
Updates #12290 
Updates #12326

### Before

image

### After

image

### CODE REVIEW

File: doc/en/_static/pytest.css [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/* Hide list bullets and top-level indent in TOC sidebar */
+.sidebar-scroll ul {list-style: none;}
+.sidebar-scroll > ul {padding-left: 0;}
+.sidebar-scroll li {margin: 0.4em 0;}
```
Comment (ferdnyc) [inline doc/en/_static/pytest.css:4]: A slight change from @pradyunsg's #12575, as I discovered that the navigation sidebar _isn't_ styled `.sidebar` (surprisingly), but the events sidebar **is**. So, nesting these changes under `.sidebar` would've hit the wrong part of the docs.

The navigation content-list is under a div at `.sidebar-drawer > .sidebar-container > .sidebar-scroll`.

I suppose I could've used either of those parent selectors instead, except for the `.sidebar-scroll > ul` change.

**Comment by ferdnyc** [comment]
I added an additional commit with a small style fix, as IMHO this looks extremely unfortunate:

image

With the added margin it becomes this, which is perfect:

image

**Review by nicoddemus — APPROVED** [review]
Thanks a lot @ferdnyc, we greatly appreciate you following up here. ❤️ 

I have skimmed through the docs and everything looks great to me. 👍

File: doc/en/_static/pytest.css (hunk 1 above)
Comment (pradyunsg) [reply doc/en/_static/pytest.css:4]: Wait, this was not meant for TOC sidebar at all. It's meant for the literal "sidebar" thing in reStructuredText: https://pradyunsg.me/furo/kitchen-sink/generic/#sidebar

File: doc/en/_static/pytest.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (pradyunsg) [inline doc/en/_static/pytest.css:4]:
```suggestion
.sidebar ul {
   list-style: none;
   padding-left: 0;
}
.sidebar ul li {
    margin: 0.4em 0;
}
```

File: doc/en/_static/pytest.css (hunk 1 above)
Comment (pradyunsg) [reply doc/en/_static/pytest.css:4]: My understanding is that folks didn't like the extra indentation that having list items gave their custom sidebar.

**Comment by pradyunsg** [comment]
I'm only now noticing the completely custom sidebar in pytest's documentation. I'll file a separate issue/PR about that, I guess.

**Comment by pradyunsg** [comment]
I've filed https://github.com/pytest-dev/pytest/pull/12584 for that.

File: doc/en/_static/pytest.css (hunk 1 above)
Comment (ferdnyc) [reply doc/en/_static/pytest.css:4]: Oh, no, pytest devs _want_ list markers in the event sidebar. The original CSS this replaces was intended to remove them only from the TOC/navigation sidebar, but it was overzealous by a lot.

Comment (pradyunsg) [reply doc/en/_static/pytest.css:4]: Oh, fun.

**Comment by pradyunsg** [comment]
FWIW, I don't consider my PR to be "competing" with this one and have no concerns with this landing -- I'm happy to rebase mine once this one goes through. I realise that might not have been clear with my previous two comments. 😅

File: doc/en/_static/pytest.css (hunk 2 above)
Comment (pradyunsg) [reply doc/en/_static/pytest.css:4]: I'm gonna say this isn't necessary because I clearly misunderstood what the goals were here.

**Comment by nicoddemus** [comment]
@ferdnyc @pradyunsg gentle ping

**Comment by ferdnyc** [comment]
@nicoddemus @pradyunsg 

To be honest, I thought this was already handled with the other PRs  that have already been merged. I'll take a look at the current state of the docs and see if anything here is still needed.

**Comment by ferdnyc** [comment]
Hmm. Looking at https://docs.pytest.org/en/latest/, list bullets are already back in the main content. They're not shown in the trainings & events sidebar, but the excessive indentation that was being caused when they were hidden is also gone, and I'm not sure my changes here would affect that anyway. 

And the main/navigation sidebar is completely changed from the previous configs, so my changes there no longer have any purpose.

As for the behind-the-scenes changes, the CSS is already moved from `_templates/style.html` to `_static/pytest-custom.css`.

I'm not sure there's anything left to do in this PR. All of the linked issues/PRs are already closed/merged. I think it's best to just close this one; everything it was trying to do is handled already.

**Comment by nicoddemus** [comment]
Thanks @ferdnyc!

**Pull request closed without merging**